### ISSUE TITLE
add style guide

### ISSUE THREAD

**Issue by wdduncan** [GenomicsStandardsConsortium/mixs#436]
It would be helpful if had a style guide for terms names (long names and short names); e.g., consistency in capitalization, use of underscores.

cc @turbomam @ramonawalls @lschriml @cmungall @only1chunts

**Comment by turbomam** [comment]
Thanks @wdduncan and @only1chunts. I have a request that came to mind while reading this issue, but isn't intended to single out any particular person or people, because we're all "guilty" of this.

Could we please refer to the attributes of MIxS terms by the column headers from the Google Sheets, or by the corresponding LinkiML attributes. I don't expect that many of us know the relevant LinkML attribute names yet, so I will be adding those to a proposed style guide, which I will link here very soon.

In the case of this issue so far, I think we're talking about 
- MIxS `Structured comment name`s, which are mapped to LinkML slot `name`s
- MIxS `Item (rdfs:label)`s and `Package item`s, which are mapped to LinkML `title`s

Alternatively, we could change the column headers in the Google Sheets to whatever people are saying most commonly, but I sure hope we can retire the Google Sheets after the QC that we are doing now!

**Comment by only1chunts** [comment]
I agree in principle we should all be using the same nomenclature for things! This has been something of an issue over the years anyway with items = terms = attributes, and has been compounded slightly more by the addition of the new terminology being added in the LinkML parlance (slots)
The short names have always been referred to as structured comment names since that is exactly what the INSDC called them, so changing that to "slots" or "slot names" is not ideal, and I would be in favour of removing any usage of the word "slot" from conversations as it seems to have multiple connotations depending whether you're talking about terms or packages.


I think we do need to clearly define the vocabulary of things somewhere and then encourage everyone involved to adopt it ASAP.
These are the current column headings in the Google sheet, with my definition of what I believe they mean (open to discussion):

- **Environmental package** - The name of the package to which an **item** belongs
- **Structured comment name** - The **item** short name, the phrase strucutred comment name is derived from the INSDC where the metadata provided by submitters is held in the Structured comment field of the INSDC entries. This has historically been the primary identifier of the **item**.
- **Package item** - This is the name of the **item** I believe this column should be renamed to **item name** it is the full length human readable name of the item without restriction on length and white space (which the structure comment name has).
- **Definition** - This is the definition/description of the **item**.
- **Expected value** - This is the type of value that is expected to be given by the submitter.
- **Value syntax** - This is the format that is expected to be used
- **Example** - an example value can be provided
- **Requirement** - This is used to indicate what the GSC expectation is on whether a value is to be provided, e.g. mandatory for compliance, mandatory IF its available (e.g. for certain sample types it may not be an appropriate measurement), recomended, optional.)
- **Preferred unit** - The unit(s) that GSC recomend using for this **item**
- **Occurrence** - How many occurances of this **item** are permitted, 1 or multiple.
- **MIXS ID** - The imutable ID assigned to the **item**

In addition we use the terminology:

- **Item** - This is how we currently refer to the thing that is represented by a specific **structured comment name** personally I prefer to call these **attributes**, many people use the word **terms** and the LinkML refer to them as **slots**.
- **package** - These are the defined collections of **items** sometimes also referred to as **Environmental package** as they are mostly specific to a given environment.
- **Checklist** - The GSC MIxS is a compilation of multiple checklists that are specific to source of DNA sequenced in an experiment, e.g. bacterial, viral, metagenomic, genomic etc... Each has its own slightly different set of  mandatory **items** , all **checklists** can be combined with any of the **packages**

**Comment by turbomam** [comment]
Those are really useful annotations, @only1chunts 

I did something similar but [more structural](https://docs.google.com/spreadsheets/d/1fU6G5gEiJSmI6ToPIhD8_Br4AqhFA9a_57yK1_7ZoKc/edit?usp=sharing) here

**Comment by only1chunts** [comment]
I've tagged this ticket to remind me to update the gensc.org website with definitions for each of the other vocabulary terms listed above

**Comment by cmungall** [comment]
It looks like we are now discussing two things here:

1. Standard nomenclature for the MIxS metamodel
2. Naming conventions for labels and structured comment names (original comment)

It looks like 1 is done.

For 2, I suggest we take @only1chunts' comments [here](https://github.com/GenomicsStandardsConsortium/mixs/issues/436#issuecomment-1190285986) and add them to the markdown docs, the close this issue
